pkgs/by-name/sk/skia-aseprite/update.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env nix-shell
+#!nix-shell --pure -i bash
+#!nix-shell -p cacert curl git jq nix-prefetch-git
+# shellcheck shell=bash
+# vim: set tabstop=2 shiftwidth=2 expandtab:
+set -euo pipefail
+shopt -s inherit_errexit
+
+[ $# -gt 0 ] || {
+  printf >&2 "usage: $0 <deps>"
+  exit 1
+}
+
+pkgpath=$(git rev-parse --show-toplevel)/pkgs/by-name/sk/skia-aseprite
+depfilter=$(tr ' ' '|' <<< "$*")
+depfile=$pkgpath/deps.nix
+pkgfile=$pkgpath/package.nix
+
+update_deps() {
+  local skiarev deps third_party_deps name url rev prefetch
+
+  version=$(sed -n 's|.*version = "\(.*\)".*|\1|p' < "$pkgfile")
+  deps=$(curl -fsS https://raw.githubusercontent.com/aseprite/skia/$version/DEPS)
+  third_party_deps=$(sed -n 's|[ ",]||g; s|:| |; s|@| |; s|^third_party/externals/||p' <<< "$deps")
+
+  printf '{ fetchgit }:\n{\n'
+  while read -r name url rev; do
+    printf >&2 'Fetching %s@%s\n' "$name" "$rev"
+    prefetch=$(nix-prefetch-git --rev "$rev" "$url")
+
+    cat << EOF
+  $name = fetchgit {
+    url = "$url";
+    rev = "$rev";
+    hash = $(jq '.hash' <<< "$prefetch");
+  };
+EOF
+
+    printf >&2 -- '----------\n\n'
+  done < <(grep -E -- "$depfilter" <<< "$third_party_deps")
+  printf '}\n'
+}
+
+update_version() {
+  local newver newrev
+  newver=$(curl -fsSL -H 'Accept: application/vnd.github+json' https://api.github.com/repos/aseprite/skia/releases/latest \
```

Comment:
GitHub API has requests limits, it's good practice to include GITHUB_TOKEN to avoid issues.

One example, from another package:
> LATEST_TAG=$(curl ${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} --silent https://api.github.com/repos/fluxcd/flux2/releases/latest | jq -r '.tag_name')

https://github.com/NixOS/nixpkgs/blob/5e9fc51b5be0f0c11cab93f69765f8e6a5e011fc/pkgs/by-name/fl/fluxcd/update.sh#L10